src/core/openthread.cpp

```diff
@@ -1322,6 +1342,20 @@ ThreadError otThreadStop(otInstance *aInstance)
     return error;
 }
 
+void otThreadSetAutoStart(otInstance *aInstance, bool aStartAutomatically)
+{
+    uint8_t autoStart = aStartAutomatically ? 1 : 0;
+    otPlatSettingsSet(aInstance, kKeyThreadAutoStart, &autoStart, sizeof(autoStart));
+}
+
+bool otThreadGetAutoStart(otInstance *aInstance)
+{
+    uint8_t autoStart = 0;
+    uint16_t autoStartLength = sizeof(autoStart);
+    otPlatSettingsGet(aInstance, kKeyThreadAutoStart, -1, &autoStart, &autoStartLength);
```

Comment:
Suggestion: It's be safer to explicitly handle the error cases from `otPlatSettingsGet()` (not trusting that `autoStart` value remains unchanged in case of a failure).